src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas2/Cas2ApplicationTest.kt

```diff
@@ -304,6 +373,30 @@ class Cas2ApplicationTest : IntegrationTestBase() {
         }
       }
     }
+
+    /**
+     * Returns true if the list of application summaries is sorted by descending created_at
+     * or false if not.
+     *
+     * Works by calculating the difference in seconds between two dates and using the sign
+     * of this difference.  If two dates are descending then the difference will be positive.
+     * If two dates are ascending the difference will be negative (which is set to 0).
+     *
+     * For a list of dates, the cumulative multiple of these signs will be 1 if all
+     * dates in the range are descending (= 1 x 1 x 1 etc.).
+     *
+     * If any dates are ascending the multiple will be 0 ( = 1 x 1 x 0 etc.).
+     *
+     * If all dates are ascending the multiple will also be 0 ( = 1 x 0 x 0 etc.).
+     */
+    private fun isCreatedAtDescending(responseBody: List<Cas2ApplicationSummary>): Boolean {
```

Comment:
@Gweaton is this better?  Added a private function and some comments.  I slightly changed the logic, but generally this is what I would have used to check something like this.  Let me know if this kind of pattern is out of sorts with what's normally done.

Reply:
I think encapsulating this in a function is much clearer, thanks. I'd probably call it `isOrderedByCreatedAtDescending` to be even more explicit to future readers - does that sound good to you?

Reply:
Yes I agree.  Makes it clearer.  Changed this now.